SP-11504: add limit to the 'sar_in' queue

przemyslawfierek-silvair opened the pull request.

przemyslawfierek-silvair committed:
SP-11568: Add limit to the 'sar_in' queue

michallowasrzechonek-silvair commented:
I'd rather fix the problem

przemyslawfierek-silvair commented:
Are you aware that the solution is the modification of the SAR mechanism (rewrite it) to reduce the number of used FD descriptors? AFAIK all features that we provide within the GW do not use the SAR mechanism at all (correct me if I'm wrong), so there is no business requirement to have this done (now). Let's talk about that on Monday.

michallowasrzechonek-silvair commented:
If so, then maybe just drop all SAR frames altogether?

przemyslawfierek-silvair commented:
I've tried and it works so this is an option.

michallowasrzechonek-silvair commented on mesh/net.c:
```diff
@@ -1959,8 +1960,18 @@ static bool seg_rxed(struct mesh_net *net, bool frnd, uint32_t iv_index,
 
 		l_debug("RXed (new: %04x %06x size: %d len: %d) %d of %d",
-				seqZero, seq, size, len, segO, segN);
+			seqZero, seq, size, len, segO, segN);
```
Formatting?

michallowasrzechonek-silvair commented:
For the record: we've decided *not* to drop all SAR transactions, since they are useful when remote access is used by support users - particularly by https://github.com/homersoft/ceiling

przemyslawfierek-silvair committed:
SP-11568: Optimize SAR message timeout in BlueZ for performance

Before this commit, BlueZ retained fully completed SAR messages in the
queue for 60 seconds, imposing a significant cost in terms of FD. With
the introduction of a limit to the SAR IN queue, it became evident that
there was room for optimization.

Refactor SAR message timeout logic:

- Timeout set to 60s for incomplete SAR messages from the receipt of the first segment.
- For complete SAR messages, timeout is 5s post-reception of the last segment.
- Affects SAR message processing performance and the number of open FDs.
